pkg/build/api/util.go

```diff
@@ -45,3 +50,25 @@ func SourceType(source BuildSource) string {
 	}
 	return sourceType
 }
+
+// LabelValue returns a string to use as a value for the Build
+// label in a pod. If the length of the string parameter exceeds
+// the maximum label length, the value will be truncated.
+func LabelValue(name string) string {
+	if len(name) <= validation.DNS1123LabelMaxLength {
+		return name
+	}
+	return name[:validation.DNS1123LabelMaxLength]
+}
+
+// GetBuildName returns the name of a Build associated with the
+// given Pod.
+func GetBuildName(pod *kapi.Pod) string {
+	return pod.Annotations[BuildAnnotation]
+}
+
+// GetBuildConfigName returns the name of a BuildConfig associated
+// with the given Build.
+func GetBuildConfigName(build *Build) string {
+	return build.Annotations[BuildConfigAnnotation]
+}
```

Comment:
(same thing for GetBuildName, unless we were already always adding the BuildAnnotation to pods?)